### ISSUE TITLE
Add support for mini-batch gradient descent

### ISSUE THREAD

**Issue by jzhang27143** [jzhang27143/federated_rc#2]
Currently the client training in src/client_utils.py: client_train_MBGD only supports batch_sizes of 1 i.e. stochastic gradient descent. This is because the gradient update code can only handle one (image, label) pair and does not average gradient updates over small batches.

Task: Modify the code under the "# gradient update" comment to average computed gradients over mini-batches.
